src/main/java/com/epam/controller/AuthenticationController.java

```diff
@@ -0,0 +1,71 @@
+package com.epam.controller;
+
+import com.epam.dto.trainee.TraineeRegistrationRequest;
+import com.epam.dto.trainee.TraineeRegistrationResponse;
+import com.epam.dto.trainer.TrainerRegistrationRequest;
+import com.epam.dto.trainer.TrainerRegistrationResponse;
+import com.epam.model.Trainee;
+import com.epam.model.Trainer;
+import com.epam.service.AuthenticationService;
+import com.epam.service.LoginAttemptService;
+import com.epam.service.TraineeService;
+import com.epam.service.TrainerService;
+import com.epam.utils.UserUtils;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.LockedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/v1/auth")
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final TrainerService trainerService;
+    private final TraineeService traineeService;
+    private final LoginAttemptService loginAttemptService;
+
+    @PostMapping("/register-trainee")
```

Comment:
/trainee/register

Reply:
###
POST http://localhost:8080/api/v1/auth/register-trainee
Content-Type: application/json

{
  "firstname": "Andrei",
  "lastname": "Pushkin",
  "dateOfBirth": "05.02.1990",
  "address": "Aldebaran"
}
Such a request returns an error 500 with no explanation that could be  understood by caller (me). Of course I have access to a log, but imagine that I have no access. What's wrong with my request to register?